build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -1587,6 +1635,18 @@ message SymlinkAbsolutePathStrategy {
   }
 }
 
+// Compression formats which may be supported.
+message Compressor {
+  enum Value {
+    // No compression. Servers and clients MUST always support this, and do
+    // not need to advertise it.
+    IDENTITY = 0;
+
+    // Zstandard compression.
+    ZSTD = 1;
```

Comment:
Is it worth adding gzip to this? I assume we expect that zstd is basically better but gzip is so commonly supported (and the protocol will tend to the superset of possibilities, similar to hash functions).

Reply:
@EricBurnett mentioned some experiments that were done with gzip, and that it was quite bad for on-the-fly tasks as required by this API (eg verifying uploads). So my preference is to defer adding gzip (or other compressors) until someone finds a scenario where it would be worthwhile.

I wonder if it's worth adding a note about the requirements for adding new compressors?

Reply:
I'd be inclined to defer it as well. If anyone (including yourself, @peterebden ) wants to request it out of intention to actually use/support gzip, we should definitely add it. But until we have signal that _someone_ wants to use it, I'd leave it off as a (slight) pressure on everyone to choose zstd from the start - uniformity is a win, and performance is a win.

From what I've seen/tested zstd is strictly better than gzip for all relevant use-cases. Only dimension where gzip _might_ win is language support, if there are REAPI implementations out there in languages without zstd implementations. But I've trivially found zstd implementations or C bindings for all languages I've spot-checked, so I don't think even that applies. Which means if you have to plug in one compressor or another, prefer to plug in zstd from the start.

W.r.t. "requirements" for new compressors, there really aren't many. It's akin to digest functions - if someone specifically wants to use something (and presumably has benchmarked to validate it'll suffice for their purposes), we should allow them to include enums for it. But we would definitely prefer everyone to standardize on the "common" ones where possible - SHA256 for digest, zstd for compression.

Reply:
OK fair enough - makes sense!